src/solver/infeasible-problem-analysis/watched-constraints.cpp

```diff
@@ -159,17 +164,22 @@ std::string HydroProduction::infeasibilityCause()
 ConstraintsFactory::ConstraintsFactory()
 {
     regex_to_ctypes_ = {
-      {"::hourly::", std::make_unique<HourlyBC, const std::string&, const double>},
-      {"::daily::", std::make_unique<DailyBC, const std::string&, const double>},
-      {"::weekly::", std::make_unique<WeeklyBC, const std::string&, const double>},
-      {"^FictiveLoads::", std::make_unique<FictitiousLoad, const std::string&, const double>},
-      {"^AreaHydroLevel::", std::make_unique<HydroLevel, const std::string&, const double>},
-      {"^Level::", std::make_unique<STS, const std::string&, const double>},
-      {"^HydroPower::", std::make_unique<HydroProduction, const std::string&, const double>}};
+      {"::hourly::", std::make_unique<HourlyBC, const std::string&, double>},
+      {"::daily::", std::make_unique<DailyBC, const std::string&, double>},
+      {"::weekly::", std::make_unique<WeeklyBC, const std::string&, double>},
+      {"^FictiveLoads::", std::make_unique<FictitiousLoad, const std::string&, double>},
+      {"^AreaHydroLevel::", std::make_unique<HydroLevel, const std::string&, double>},
+      {"^Level::", std::make_unique<STS, const std::string&, double>},
+      {"^HydroPower::", std::make_unique<HydroProduction, const std::string&, double>},
+      {"^WithdrawalSum::", std::make_unique<STSWithdrawalSum, const std::string&, double>},
+      {"^InjectionSum::", std::make_unique<STSInjectionSum, const std::string&, double>},
```

Comment:
while this syntax is functional (function as value = std::make_unique) it's hard to read. why not be more explicit by replacing it with lambda and inside it a call to std::make_unique<T>(..) ?

Reply:
Yes I agree it's a bit hard to read as it is, but when taking a look at the rest of the code it's kind of obvious it is a function (`std::function`)